engine/tests/phpunit/integration/Elgg/Actions/ActionRegistrationTest.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+namespace Elgg\Actions;
+
+use Elgg\IntegrationTestCase;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * @group ActionsService
+ */
+class ActionRegistrationTest extends IntegrationTestCase {
+
+	/**
+	 * Skip some actions that do not play well with testing suite
+	 * @var array
+	 */
+	private $skips = [
+		'logout',
+	];
+
+	public function up() {
+		// Logging in admin so all actions are accessible
+		_elgg_services()->session->setLoggedInUser($this->getAdmin());
+	}
+
+	public function down() {
+		_elgg_services()->session->removeLoggedInUser();
+	}
+
+	public function actionsProvider() {
+		self::createApplication();
+
+		$provides = [];
+
+		$actions = _elgg_services()->actions->getAllActions();
+
+		foreach ($actions as $name => $params) {
+			$provides[] = [
+				$name,
+				$params['file'],
+				$params['access']
+			];
+		}
+
+		return $provides;
+	}
+
+	/**
+	 * @dataProvider actionsProvider
+	 */
+	public function testCanRequestActionWithoutParameters($name, $file, $access) {
+
+		if (in_array($name, $this->skips)) {
+			$this->markTestSkipped("Can not test action '$name'");
+		}
+
+		$request = $this->prepareHttpRequest("action/$name", 'POST', [], false, true);
+
+		$this->assertTrue(_elgg_services()->router->route($request));
```

Comment:
@hypeJunction scrutinizer is not detecting that these actions are processed/executed... is this working as intended? The login actions is done a bit different... and that action is now correctly covered... i was expecting that the other actions would be covered a little bit by this test

Reply:
I have been noticing that the coverage is not very accurate. E.g. with language files you would expect 100% coverage, because they just return arrays. I will try to generate a coverage report locally to see if it's the xdebug version on scrutinizer.